staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -290,6 +291,31 @@ func (r *crdHandler) updateCustomResourceDefinition(oldObj, newObj interface{})
 	r.customStorageLock.Lock()
 	defer r.customStorageLock.Unlock()
 
+	// This can't work like this, but I just want to try something. Before there was
+	// some additional logic, but I got rid of it before pushing the commit.
+	// So, this is called when CRD is updated. That includes creation, but termination as well.
+	// Probably some logic like in the naming controller, but only to fire state change
+	// if CRD is creating for the first time.
+	if !apiextensions.IsCRDConditionTrue(newCRD, apiextensions.Established) {
+		establishedCondition := apiextensions.CustomResourceDefinitionCondition{
+			Type:    apiextensions.Established,
+			Status:  apiextensions.ConditionTrue,
+			Reason:  "InitialNamesAccepted",
+			Message: "the initial names have been accepted",
+		}
+		// This could miss some logic, like applying changes (present in the naming controller)
+		// Also, I'm supposed to update the newCRD, right?
+		apiextensions.SetCRDCondition(newCRD, establishedCondition)
+
+		// This is used to updated the server. The above `SetCRDCondition` only sets in storage.
+		//crdClient, err := internalclientset.NewForConfig(s.GenericAPIServer.LoopbackClientConfig)
```

Comment:
this client should be created once outside of the crd handler and then passed to NewCustomResourceDefinitionHandler.